pkg/builder/vm.go

```diff
@@ -0,0 +1,227 @@
+	}
+	objectMeta := metav1.ObjectMeta{
+		Namespace:    defaultVMNamespace,
+		GenerateName: defaultVMGenerateName,
+		Labels:       vmLabels,
+		Annotations:  map[string]string{},
+	}
+	running := pointer.BoolPtr(false)
+	cpu := &kubevirtv1.CPU{
+		Cores: defaultVMCPUCores,
+	}
+	resources := kubevirtv1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceMemory: resource.MustParse(defaultVMMemory),
+		},
+	}
+	template := &kubevirtv1.VirtualMachineInstanceTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: vmLabels,
+		},
+		Spec: kubevirtv1.VirtualMachineInstanceSpec{
+			Domain: kubevirtv1.DomainSpec{
+				CPU: cpu,
+				Devices: kubevirtv1.Devices{
+					Disks:      []kubevirtv1.Disk{},
+					Interfaces: []kubevirtv1.Interface{},
+				},
+				Resources: resources,
+			},
+			Affinity: &corev1.Affinity{},
+			Networks: []kubevirtv1.Network{},
+			Volumes:  []kubevirtv1.Volume{},
+		},
+	}
+
+	vm := &kubevirtv1.VirtualMachine{
+		ObjectMeta: objectMeta,
+		Spec: kubevirtv1.VirtualMachineSpec{
+			Running:             running,
+			Template:            template,
+			DataVolumeTemplates: []kubevirtv1.DataVolumeTemplateSpec{},
+		},
+	}
+	return &VMBuilder{
+		VirtualMachine: vm,
+	}
+}
+
+func (v *VMBuilder) Name(name string) *VMBuilder {
+	v.VirtualMachine.ObjectMeta.Name = name
+	v.VirtualMachine.ObjectMeta.GenerateName = ""
+	v.VirtualMachine.Spec.Template.ObjectMeta.Labels[LabelKeyVirtualMachineName] = name
+	return v
+}
+
+func (v *VMBuilder) Namespace(namespace string) *VMBuilder {
+	v.VirtualMachine.ObjectMeta.Namespace = namespace
+	return v
+}
+
+func (v *VMBuilder) MachineType(machineType string) *VMBuilder {
+	v.VirtualMachine.Spec.Template.Spec.Domain.Machine.Type = machineType
+	return v
+}
+
+func (v *VMBuilder) HostName(hostname string) *VMBuilder {
+	v.VirtualMachine.Spec.Template.Spec.Hostname = hostname
+	return v
+}
+
+func (v *VMBuilder) Description(description string) *VMBuilder {
+	if v.VirtualMachine.ObjectMeta.Annotations == nil {
+		v.VirtualMachine.ObjectMeta.Annotations = map[string]string{}
+	}
+	v.VirtualMachine.ObjectMeta.Annotations[AnnotationKeyDescription] = description
+	return v
+}
+
+func (v *VMBuilder) Labels(labels map[string]string) *VMBuilder {
+	if v.VirtualMachine.ObjectMeta.Labels == nil {
+		v.VirtualMachine.ObjectMeta.Labels = labels
+	}
+	for key, value := range labels {
+		v.VirtualMachine.ObjectMeta.Labels[key] = value
+	}
+	return v
+}
+
+func (v *VMBuilder) Annotations(annotations map[string]string) *VMBuilder {
+	if v.VirtualMachine.ObjectMeta.Annotations == nil {
+		v.VirtualMachine.ObjectMeta.Annotations = annotations
+	}
+	for key, value := range annotations {
+		v.VirtualMachine.ObjectMeta.Annotations[key] = value
+	}
+	return v
+}
+
+func (v *VMBuilder) Memory(memory string) *VMBuilder {
+	v.VirtualMachine.Spec.Template.Spec.Domain.Resources.Requests = corev1.ResourceList{
+		corev1.ResourceMemory: resource.MustParse(memory),
```

Comment:
Is it ok if `resource.MustParse` panics with an invalid input?

Reply:
yes